### PR TITLE
Add index to primary vertex in scouting particle collection

### PR DESCRIPTION
**Pull request by davidsheffield**
Backport of #11029.

### CODE REVIEW

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
+1

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
+1

File: DataFormats/Scouting/interface/ScoutingParticle.h [hunk 1]
```diff
@@ -10,10 +10,10 @@ class ScoutingParticle
     public:
         //constructor with values for all data fields
         ScoutingParticle(float pt, float eta, float phi, float m, 
-                int pdgId):
-            pt_(pt), eta_(eta), phi_(phi), m_(m), pdgId_(pdgId) {}
+		 int pdgId, int vertex):
+            pt_(pt), eta_(eta), phi_(phi), m_(m), pdgId_(pdgId), vertex_(vertex) {}
         //default constructor
-        ScoutingParticle():pt_(0), eta_(0), phi_(0), m_(0), pdgId_(0) {}
+        ScoutingParticle():pt_(0), eta_(0), phi_(0), m_(0), pdgId_(0), vertex_(0) {}
```
Comment (Dr15Jones) [inline DataFormats/Scouting/interface/ScoutingParticle.h]: Given 0 is a valid index, how about using max int instead? That way reading old data one can spot if the variable is not filled.

Comment (fwyzard) [inline DataFormats/Scouting/interface/ScoutingParticle.h]: Or -1 ?

Comment (davidsheffield) [inline DataFormats/Scouting/interface/ScoutingParticle.h]: @Dr15Jones, you mean changing it to `vertex_(2147483647)`?

-1 is a valid value for this member. The producer sets it to -1 when no vertex is matched.

Comment (Dr15Jones) [inline DataFormats/Scouting/interface/ScoutingParticle.h]: Given when reading old data there would be no match, then -1 makes the most sense.

**Comment by davidsheffield** [comment]
I got the same checksum from scram b for the class as before the most recent commit despite changing the code. I assume that's wrong. How should I get the checksum?

**Comment by davidlange6** [comment]
@davidsheffield  -the checksum is derived from the member data not the c++ that fills them. So all is ok.

**Comment by davidlange6** [comment]
@cmsbuild, please test

**Comment by smuzaffar** [comment]
comparison is not ready as jenkins job failed to upload the result to cmssdt server. I have restarted the comparison.

**Pull request merged**